candy-machine/program/src/utils.rs

```diff
@@ -71,6 +72,13 @@ pub struct TokenTransferParams<'a: 'b, 'b> {
     pub token_program: AccountInfo<'a>,
 }
 
+pub fn punish_bots(err: ErrorCode, bot_account: &AccountInfo, payment_account: &AccountInfo, fee: u64) -> Result<(), ProgramError> {
+    msg!("Error: {}, Candy Machine Botting is taxed at {:?} lamports", err.to_string(), fee);
+    **bot_account.try_borrow_mut_lamports()? = payment_account.lamports() - fee;
+    **payment_account.try_borrow_mut_lamports()? = payment_account.lamports() + fee;
```

Comment:
prolly want checked math?

Reply:
oh yeah, and if there's only `fee - 1` in the account then drain that instead of erroring out